src/core/thread/network_data_tlvs.hpp

```diff
@@ -505,7 +512,8 @@ class BorderRouterEntry
      */
     void SetPreference(int8_t aPrf)
     {
-        mFlags = (mFlags & ~kPreferenceMask) | ((static_cast<uint8_t>(aPrf) << kPreferenceOffset) & kPreferenceMask);
+        mFlags = HostSwap16((HostSwap16(mFlags) & ~kPreferenceMask) |
+                            ((static_cast<uint16_t>(aPrf) << kPreferenceOffset) & kPreferenceMask));
```

Comment:
nit: use `GetFlags()` here to replace the `(HostSwap16(mFlags) & ~kPreferenceMask)`?

Reply:
updated.